fix(text-input-group): updates to fix when used in dark block

mcoker opened the pull request:
fixes https://github.com/patternfly/patternfly/issues/5670

thatblindgeye reviewed (commented):
Only thing I'm noticing is when dark mode is enable via the switch and then applying `pf-v5-t-light` to `.pf-v5-c-text-input-group`, the icon color becomes black against gray input background. Similar occurrence when applying the `pf-v5-t-light` class to the root when the dark mode switch is off.

Depends how likely a use case that is, but otherwise things look good from a quick look at the preview

mcoker commented:
@thatblindgeye great catch! Definitely something we should fix. I needed to reset the pf-v5-t-[light/dark] vars in dark theme for the icon vars I added in the previous commit. I also found a block of var declarations that I'm pretty sure we don't need in [sass-utilities/themes/dark/placeholders.scss](https://github.com/patternfly/patternfly/blob/2266ace7039dd918193166282d42ea66f22ab109/src/patternfly/sass-utilities/themes/dark/placeholders.scss), so I removed them, but take a look around and see if you notice anything off. I imagine these are the places we would see issues:

```
cmichael-mac:components cmichael$ grep -ri pf-v5-t- *
AboutModalBox/about-modal-box.scss:  @include pf-v5-t-dark;
Accordion/accordion.scss:  @include pf-v5-t-light;
Alert/alert.scss:  @include pf-v5-t-light; // This component always needs to be light
Banner/banner.scss:  @include pf-v5-t-dark;
Banner/banner.scss:    @include pf-v5-t-light(false);
Banner/banner.scss:    @include pf-v5-t-light(false);
CalendarMonth/calendar-month.scss:  @include pf-v5-t-light;
Chip/chip-group.scss:  @include pf-v5-t-light; // force the container follow the light theme
Chip/chip.scss:  @include pf-v5-t-light;  // This component always needs to be light
ContextSelector/context-selector.scss:  @include pf-v5-t-light;
DataList/data-list.scss:  @include pf-v5-t-light;
FormControl/form-control.scss:  @include pf-v5-t-light("--#{$form-control}--Color");
LogViewer/log-viewer.scss:      @include pf-v5-t-dark;
Login/login.scss:  @include pf-v5-t-dark;
Login/login.scss:  @include pf-v5-t-dark;
Masthead/masthead.scss:  @include pf-v5-t-dark; // force the container to follow the dark theme
Masthead/masthead.scss:    @include pf-v5-t-light;
Masthead/masthead.scss:    @include pf-v5-t-light;
Masthead/masthead.scss:    @include pf-v5-t-dark(--#{$button}--m-plain--Color); // for all plain buttons, especially inside other components (.pf-v5-c-input-group), for dark mode
Menu/menu.scss:  @include pf-v5-t-light;
NotificationBadge/examples/NotificationBadge.md:<div class="pf-v5-t-dark">
NotificationBadge/examples/NotificationBadge.md:<div class="pf-v5-t-dark">
NotificationBadge/examples/NotificationBadge.md:<div class="pf-v5-t-dark">
Page/page.scss:  @include pf-v5-t-dark; // force the container to follow the dark theme
Page/page.scss:  @include pf-v5-t-dark;
Page/page.scss:    @include pf-v5-t-light;
Page/page.scss:      @include pf-v5-t-dark;
Page/page.scss:    @include pf-v5-t-dark; // force the container to follow the dark theme
Select/select.scss:  @include pf-v5-t-light; // force the container follow the light theme
Table/table.scss:  @include pf-v5-t-light; // This component always needs to be light
Wizard/wizard.scss:  @include pf-v5-t-dark;
```

thatblindgeye reviewed (approved):
🎉 looks good from what I can tell. One other question is whether `pf-v5-t-light` should override the root theme? If the theme switcher is set to light theme and I apply the dark theme class to the TextInputGroup, it styles that component with its dark theme. If the theme switcher is set to dark, though, applying the light theme class doesn't style the component with its light theme.

This isn't just for TextInputGroup, though, as it looks like the Navigation examples showcasing light theme have the same issue.

srambach commented on src/patternfly/base/_variables.scss:
```diff
@@ -232,6 +232,10 @@
   // Icons
   --#{$pf-global}--icon--Color--light: #{$pf-v5-global--icon--Color--light};
   --#{$pf-global}--icon--Color--dark: #{$pf-v5-global--icon--Color--dark};
+  --#{$pf-global}--icon--Color--light--light: #{$pf-v5-global--icon--Color--light--light};
```
I would very much appreciate a comment clarifying which variable is which (e.g. --[light/dark component]--[light/dark theme] or whatever it means :-)

srambach commented:
Did this break the horizontal nav of the workspace? Or is this PR missing a different fix? 
<img width="1365" alt="image" src="https://github.com/patternfly/patternfly/assets/19825616/69057d45-e2ca-4661-a200-1fed6cdb31ce">

thatblindgeye commented:
@srambach that's how it currently is on the core and react v5 workspaces, too

srambach commented:
> @srambach that's how it currently is on the core and react v5 workspaces, too

Ah, my bad - I looked at another preview and thought it was ok there but I was mistaken.

srambach reviewed (approved):
Input in masthead looks good 👍

patternfly-build commented:
:tada: This PR is included in version 5.0.0-prerelease.8 :tada:

The release is available on:
- [GitHub release](https://github.com/patternfly/patternfly/releases/tag/prerelease-v5.0.0-prerelease.8)
- [npm package (@prerelease dist-tag)](https://www.npmjs.com/package/@patternfly/patternfly/v/5.0.0-prerelease.8)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

patternfly-build commented:
:tada: This PR is included in version 6.0.0-alpha.1 :tada:

The release is available on:
- [GitHub release](https://github.com/patternfly/patternfly/releases/tag/prerelease-v6.0.0-alpha.1)
- [npm package (@alpha dist-tag)](https://www.npmjs.com/package/@patternfly/patternfly/v/6.0.0-alpha.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
